library/core/src/option.rs

```diff
@@ -963,9 +963,10 @@ impl<T> Option<T> {
     /// assert_eq!(y, None);
     /// ```
     #[inline]
+    #[rustc_const_unstable(feature = "const_option", issue = "67441")]
     #[stable(feature = "rust1", since = "1.0.0")]
-    pub fn take(&mut self) -> Option<T> {
-        mem::take(self)
+    pub const fn take(&mut self) -> Option<T> {
+        mem::replace(self, None)
```

Comment:
I'm personally not a big fan of sacrificing the reuse of already existing code and using the raw implementation of the abstracting function for being able to make it `const`. Shouldn't the first step be to make `mem::take` a `const fn`? Or otherwise shouldn't there be a `TODO` added to change this back to `mem::take(self)` later?
Unless of course it's not possible at all to make `mem::take` a `const`, but I believe it is, at least at some point.

Reply:
For mem::take to be constant, we would need `impl const` to go live, and it's gated under `incomplete_feature` for now according to this (https://github.com/rust-lang/rust/pull/86808), so it did not really feel right to try to do something around that.

But I can add a TODO for sure to this!